.github/workflows/tg-model-perf-tests-impl.yaml

```diff
@@ -41,13 +48,25 @@ jobs:
         run: |
           echo "TT_METAL_HOME=$(pwd)" >> $GITHUB_ENV
           echo "PYTHONPATH=$(pwd)" >> $GITHUB_ENV
-      - uses: actions/download-artifact@v4
+      - name: Download profiler build artifact
+        id: download-profiler-artifact
+        if: ${{ matrix.test-group.tracy }}
+        uses: actions/download-artifact@v4
+        with:
+          name: TTMetal_build_${{ matrix.test-group.arch }}_profiler
+        continue-on-error: true
```

Comment:
Why does this one continue-on-error, but the non-tracy one doesn't?

Reply:
this is to check if the profiler build is there or not as the tracy flag cant ensure that this build is done(this is maily done for the choose your pipeline as it cant have both build at same time)

Reply:
Sorry I don't follow.

Doesn't this work?

```
... do stuff unconditionally
- download Tracy artifact
  if we want profiler
- download non-Tracy artifact
  if we don't want profiler
- do stuff with the tracy artifact
  if we want profiler
- do stuff with the non-tracy artifact
  if we don't want the profiler
- do common stuff
...
```
And if at any point something fails, nothing further happens.

Reply:
It will work but the issue is, assume the case where the job needs profiler build and at the same time, what happens if the profiler build artifact is not found?, it will mark the job as failure right? i wanted to avoid that failure

Reply:
If this TG perf pipeline ran separately, I have added the code to have both builds and during run it will take the necessary build it required, but in case of choose your pipeline, any one build can be done right? Either regular build or Tracy build 

Some conditions can be removed if,
When we select the Tracy build checkbox, instead of having only Tracy build, if it generates both build, then that will solve these issues. Is that possible? @tt-rkim

Reply:
Hmm, in that case, I'm wondering if a new workflow file is better instead. CCL tests don't scream model tests in my mind.

Anyway, for the current situation, for choose your pipeline, yes we can do what you say.

We can separate out the tracy branch, and build both branches if the checkbox is clicked. I'm wondering if it is helpful to keep the checkbox clicked by default, rather than unclicked.

Then for tg model perf it must require both a `build-artifact` and `build-artifact-profiler` job beforehand with the latter job having `tracy` enabled in the `choose your pipeline`. Otherwise, it will not run.

It might be good to have a message stating this, clarifying that this is now the case.

Reply:
Hey @tt-rkim - what's needed for us to have a new workflow for CCL op perf measurement here? and CCL perf in general, even beyond complete "ops"?

Reply:
Instead of having both builds for this tg model perf pipeline, can we have the Tracy build alone for entire job and make the note like this job requires Tracy build in choose your pipeline? Then we can get rid of additional conditions, will that work?

Reply:
* We could have a separate CCL workflow, so you wouldn't be touching this one anymore. Feel free to write one up in this PR if that's what you guys prefer
* You can also try only using profiler build, however I'm not sure if instrumentation would make a diff in the other perf tests. You may want to clear with the test owners (@cglagovich @tt-aho ) that this is ok